### ISSUE TITLE
question regarding system quotas

### ISSUE THREAD

**Issue by gcornacchia** [Axway-API-Management-Plus/apim-cli#362]
### APIM-CLI version

1.13.0

### API-Management version

7.7.20211130

### Question

HI,
I cannot find in the docs how can I set **system quota on all API** through apim-cli?

From what I've read we can set up system quota in API config file (releated to the single API) or we can set up application quota in application config file.

Am I missing something?

**Comment by rathnapandi** [comment]
Hi @gcornacchia, 
We can setup both quotas via api config file. If it is application specific we can do it in application config file. 

## API config file example 

```json
{
   "name":"Samples Complete API",
   "path":"/sample/complete/api/v1",
   "state":"published",
   "summary":"My complete API-Summary",
   "version":"1.0.0",
   "organization":"API Development",
   "descriptionType":"manual",
   "descriptionManual":"This is my __markdown__ based API description.",
   "vhost":"api.custom-host.com",
   "backendBasepath":"https://any.server.com:7676",
   "image": "../images/sample-api-icon.jpg", 
   "securityProfiles":[
      {
         "name":"_default",
         "isDefault":true,
         "devices":[
            {
               "name":"API Key",
               "type":"apiKey",
               "order":0,
               "properties":{
                  "apiKeyFieldName":"KeyId",
                  "takeFrom":"HEADER",
                  "removeCredentialsOnSuccess":"false"
               }
            }
         ]
      }
   ],
   "outboundProfiles":{
      "_default":{
         "requestPolicy":"Request policy 1",
         "routePolicy":"Routing policy 1",
         "responsePolicy":"Response policy 1",
         "faultHandlerPolicy":"Faulthandler policy 1"
      }
   },
   "authenticationProfiles":[
      {
         "name":"_default",
         "isDefault":"true",
         "parameters":{
            "apiKey":"4249823490238490",
            "apiKeyField":"KeyId",
            "httpLocation":"QUERYSTRING_PARAMETER"
         },
         "type":"apiKey"
      }
   ],
   "tags":{
      "tag-name 2":[
         "value 3",
         "value 4"
      ],
      "tag-name 1":[
         "value 1",
         "value 2"
      ]
   },
   "customProperties":{
      "customProperty1":"Test-Input 1",
      "customProperty2":"1",
      "customProperty3":"true"
   },
   "corsProfiles":[
      {
         "name":"_default",
         "isDefault":true,
         "origins":[
            "*"
         ],
         "allowedHeaders":[
            "Authorization"
         ],
         "exposedHeaders":[
            "via"
         ],
         "supportCredentials":false,
         "maxAgeSeconds":0
      }
   ],
   "caCerts":[
      {
         "certFile":"../certificates/swaggerio.crt",
         "useForOutbound":true,
         "useForInbound":false
      },
      {
         "certFile":"../certificates/GoDaddyRootCertificateAuthority-G2.crt",
         "useForOutbound":true,
         "useForInbound":false
      },
      {
         "certFile":"../certificates/GoDaddySecureCertificateAuthority-G2.crt",
         "useForOutbound":true,
         "useForInbound":false
      },
      {
         "certFile":"../certificates/GlobalSignRootCA-R2.crt",
         "useForOutbound":false,
         "useForInbound":true
      }
   ], 
   "applicationQuota":{
      "restrictions":[
         {
            "method":"*",
            "type":"throttlemb",
            "config":{
               "period":"hour",
               "mb":555,
               "per":1
            }
         }
      ]
   },
   "systemQuota":{
      "restrictions":[
         {
            "method":"*",
            "type":"throttle",
            "config":{
               "period":"hour",
               "messages":666,
               "per":2
            }
         }
      ]
   }
}
```

## Application Quota config

```json
{
  "name": "${appName}",
  "organization": "${orgName}",
  "state": "approved",
  "enabled": true,
  "email": "user@domain.com",
  "phone": "654654646234",
  "appQuota": {
    "restrictions": [
      {
        "api": "*",
        "method": "*",
        "type": "throttle",
        "config": {
          "messages": "1000",
          "period": "day",
          "per": "1"
        }
      }, 
      {
        "api": "${apiName}",
        "method": "*",
        "type": "throttle",
        "config": {
          "messages": "2000",
          "period": "minute",
          "per": "1"
        }
      },
      {
        "api": "${apiName}",
        "method": "addPet",
        "type": "throttle",
        "config": {
          "messages": "3000",
          "period": "hour",
          "per": "1"
        }
      }
    ]
  }
}
```

**Comment by gcornacchia** [comment]
hi @rathnapandi 
my answer is, 
with apim-cli how can I handle this system quota regarding all api (see screenshot below)?

![image](https://user-images.githubusercontent.com/15086835/223984372-9d2b6207-2ea5-42ad-8ac6-2a8985d879d7.png)

**Comment by gcornacchia** [comment]
@rathnapandi have you seen my last comment here?

**Comment by rathnapandi** [comment]
Hi @gcornacchia, 
Looks like the current code is not supporting it.  I will add this feature in next release.

**Comment by rathnapandi** [comment]
Hi @gcornacchia, 

Support of exporting and importing all on System and application quota implemented in separate module.  

Command to export all quotas 

```bash 
/apim.sh settings get -type globalquotas -o json -u apiadmin -p changeme -h 10.129.61.129
```

Command to import all quotas. 
```bash 
 ./apim.sh settings import -type globalquotas -c axway-api-manager/apimanager-config.json -u apiadmin -p changeme -h 10.129.61.12
```

Sample config file - 

```json
{
  "config" : null,
  "remoteHosts" : null,
  "alerts" : null,
  "quotas" : {
    "systemQuota" : {
      "method" : "*",
      "type" : "throttle",
      "config" : {
        "messages" : "1000",
        "period" : "second",
        "per" : "2"
      },
      "api" : "*"
    },
    "applicationQuota" : {
      "method" : "*",
      "type" : "throttle",
      "config" : {
        "messages" : "2000",
        "period" : "second",
        "per" : "2"
      },
      "api" : "*"
    }
  }
}

```

Use the attached binary to test it. 
[axway-apimcli-1.13.6-SNAPSHOT.tar.gz](https://github.com/Axway-API-Management-Plus/apim-cli/files/11072898/axway-apimcli-1.13.6-SNAPSHOT.tar.gz)

**Comment by rathnapandi** [comment]
fixed in version 1.13.6
